frontend/src/metabase/embedding-sdk/types/components/data-picker.ts

```diff
@@ -1,16 +1,21 @@
-import type { TableId } from "metabase-types/api";
+import type { CardType, TableId } from "metabase-types/api";
 
 export interface DataSourceSelectorProps {
   isInitiallyOpen: boolean;
-  /** Whether the first stage of the query uses a model */
-  isQuerySourceModel: boolean;
+  /** Type of the the query's first stage */
+  querySourceType:
+    | Extract<CardType, "model" | "question">
+    | undefined
+    // This allows the predefined values to still be there, so TypeScript doesn't reduce the type to just `string`
+    | (string & NonNullable<unknown>);
```

Comment:
Because the type passed also has more values than just `"model"` or `"question"`

https://github.com/metabase/metabase/blob/7d645a09127dd256acb6535229c4f13460753ff3/frontend/src/metabase/querying/notebook/components/NotebookDataPicker/EmbeddingDataPicker/EmbeddingDataPicker.tsx#L118